src/libtelio.udl

```diff
@@ -0,0 +1,520 @@
+[Custom]
+typedef string PublicKey;
+
+[Custom]
+typedef string SecretKey;
+
+[Custom]
+typedef string IpAddr;
+
+[Custom]
+typedef string IpNetwork;
+
+[Custom]
+typedef string Ipv4Addr;
+
+[Custom]
+typedef string SocketAddr;
+
+namespace libtelio {
+    /// Set the global logger.
+    /// # Parameters
+    /// - `log_level`: Max log level to log.
+    /// - `logger`: Callback to handle logging events.
+    void set_global_logger(TelioLogLevel log_level, TelioLoggerCb logger);
+
+    /// Get default recommended adapter type for platform.
+    TelioAdapterType uniffi_telio_get_default_adapter();
+
+    /// Get current version tag.
+    string uniffi_telio_get_version_tag();
```

Comment:
Some functions here are prefixed with `uniffi_telio_`, that is temporary and will be removed in a future PR. It's just to prevent name clashes with existing functions until those functions are removed